File: HCS.Client/ClientApp/app/models/tariff.ts
import { KeyValuePair } from "./key_value_pair";
import { DataSource } from "@angular/cdk/collections";
import { Observable } from "rxjs/Observable";
import "rxjs/add/observable/of";

export class Block {
    limit: number;
    price: number;
}

export class Tariff {
    id: number;
    name: string;
    subscriberFee: number;
    price: number;
    providedUtilityId: number;
    consumerTypeId: number;
    blocks: Block[];

    constructor() {
        this.blocks = [];
    }
}

export class TariffInfo {
    id: number;
    name: string;
    consumerType: string;
}

export class TariffListItem {
    id: number;
    position: number;
    name: string;
    providedUtility: string;
    consumerType: string;
    consumersQuantity: number;
}

export class TariffDataSource extends DataSource<any> {

    constructor(private tariffs: TariffListItem[]) {
        super();
    }

    connect(): Observable<TariffListItem[]> {
        return Observable.of(this.tariffs);
    }

    disconnect() { }
}

File: HCS.Core/Domain/Tariff.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace HCS.Core.Domain
{
    public class Tariff
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal SubscriberFee { get; set; }
        public decimal Price { get; set; }
        public int ConsumerTypeId { get; set; }
        public ConsumerType ConsumerType { get; set; }
        public int ProvidedUtilityId { get; set; }
        public ProvidedUtility ProvidedUtility { get; set; }
        public ICollection<ConsumedUtility> ConsumedUtilities { get; set; }
        public ICollection<Block> Blocks { get; set; }

        public Tariff()
        {
            Blocks = new Collection<Block>();
            ConsumedUtilities = new Collection<ConsumedUtility>();
        }
    }
}
